tests/Integration/Behaviour/Features/Context/Domain/Product/RelatedProductsFeatureContext.php

```diff
@@ -0,0 +1,121 @@
+{
+    /**
+     * @Then product :productReference should have no related products
+     *
+     * @param string $productReference
+     */
+    public function assertProductHasNoRelatedProducts(string $productReference): void
+    {
+        $productId = $this->getSharedStorage()->get($productReference);
+        try {
+            $relatedProducts = $this->getQueryBus()->handle(new GetRelatedProducts($productId));
+            Assert::assertEmpty(
+                $relatedProducts,
+                sprintf('Product %s expected to have no related products', $productReference)
+            );
+        } catch (DomainException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @When I set following related products to product :productReference:
+     *
+     * @param string $productReference
+     * @param TableNode $tableNode
+     */
+    public function setRelatedProducts(string $productReference, TableNode $tableNode): void
+    {
+        $productId = $this->getSharedStorage()->get($productReference);
+        $references = array_keys($tableNode->getRowsHash());
+        $relatedProductIds = [];
+
+        foreach ($references as $reference) {
+            $relatedProductIds[] = $this->getSharedStorage()->get($reference);
+        }
+
+        try {
+            $this->getCommandBus()->handle(new SetRelatedProductsCommand($productId, $relatedProductIds));
+        } catch (DomainException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then product :productReference should have following related products:
+     *
+     * @param string $productReference
+     * @param TableNode $tableNode
+     */
+    public function assertRelatedProducts(string $productReference, TableNode $tableNode)
+    {
+        $productId = $this->getSharedStorage()->get($productReference);
+
+        $expectedReferences = array_keys($tableNode->getRowsHash());
+        $actualRelatedProducts = $this->getQueryBus()->handle(new GetRelatedProducts($productId));
+
+        $expectedIds = array_map(function (string $reference): int {
+            return $this->getSharedStorage()->get($reference);
+        }, $expectedReferences);
+
+        $actualIds = array_map(function (RelatedProduct $relatedProduct): int {
+            return $relatedProduct->getProductId();
+        }, $actualRelatedProducts);
+
+        Assert::assertEquals($expectedIds, $actualIds, 'Unexpected related products');
+    }
+
+    /**
+     * @When I remove all related products from product :productReference
+     *
+     * @param string $productReference
+     */
+    public function removeAllRelatedProducts(string $productReference)
+    {
+        $productId = $this->getSharedStorage()->get($productReference);
+        try {
+            $this->getCommandBus()->handle(new RemoveAllRelatedProductsCommand($productId));
+        } catch (DomainException $e) {
```

Comment:
Same here, exception is too wide And if you don't expect to test the caught exception don't event catch it